tests/xdp/xdp_tests.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// This module facilitates testing various XDP scenarios by sending traffic to a remote system
+// running XDP eBPF hook and an attached XDP program.
+// For the reflection test, reflect_packet.o needs to be loaded on the remote host.
+
+#define CATCH_CONFIG_RUNNER
+
+#include "catch_wrapper.hpp"
+#include "xdp_tests.h"
+#include "xdp_tests_common.h"
+
+std::string _remote_ip;
+const uint16_t _reflection_port = REFLECTION_TEST_PORT;
+
+TEST_CASE("xdp_reflect_test", "[xdp_tests]")
+{
+    int error = 0;
+    // Create a receiver socket.
+    SOCKET receiver_socket = INVALID_SOCKET;
+    receiver_socket = WSASocket(AF_INET, SOCK_DGRAM, IPPROTO_UDP, nullptr, 0, WSA_FLAG_OVERLAPPED);
+    if (receiver_socket == INVALID_SOCKET)
+        FAIL("Failed to create receiver socket with error: " << WSAGetLastError());
+
+    // Bind it to the reflection port.
+    struct sockaddr_in recv_addr;
```

Comment:
Fix test to not be IPv4-only.  (Welcome to the 21st century :)

Reply:
https://blog.allstream.com/a-funny-thing-happened-on-the-way-to-ipv6/